### PR TITLE
feat: Multicharts

### LINKED ISSUE #949: Sub issue related to #948
# Summary

This is task of #948 
This particular tickets enables Devtron to add multiple charts so that they can used at the time of creation of application. Allowing user to upload their own charts is out of scope for this task

# Proposal

1. User should be able to select chart type if they have yet not selected any chart type
2. If user has selected and saved a chart then user should not be able to change the chart type

Screenshot 2021-12-06 at 7 44 57 PM

Screenshot 2021-12-06 at 7 44 47 PM

Screenshot 2021-12-06 at 7 44 38 PM

### PR DESCRIPTION
**Pull request by aviralsrivastava01**
# Description

Currently only one chart type is supported/visible to user in the deployment template section. Product should be able to support multiple chart types for eg knative, cronjob and others. Also user should be able to add their own helm chart for selection in deployment template.

Fixes #949 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test case A
- [ ] Test case B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas
* [ ] I have tested it for all user roles

### CODE REVIEW

**Comment by pghildiyal** [comment]
https://docs.google.com/document/d/1OkdqUes4BGMlzZvpRYZME4U76KcWaosCKuyeNUwMMbc/edit

**Comment by aviralsrivastava01** [comment]
**Done:**
User needs to create configmap with: 	
   1. Annotation:  charts.devtron.ai/data: mount
   2. Configmap data key is the name of the file and value is the data in gz format

File: api/restHandler/AppStoreRestHandler.go [hunk 1]
```diff
@@ -477,6 +477,13 @@ func (handler *AppStoreRestHandlerImpl) TriggerChartSyncManual(w http.ResponseWr
 		common.WriteJsonResp(w, err, nil, http.StatusUnauthorized)
 		return
 	}
+
+	//Chart repo sync error handling
 	TriggerResult := handler.appStoreService.TriggerChartSyncManual()
-	common.WriteJsonResp(w, nil, TriggerResult, http.StatusOK)
-}
\ No newline at end of file
+	if TriggerResult != nil {
```
Comment (manish-agrawal-ai) [inline api/restHandler/AppStoreRestHandler.go:483]: This change is not related to this story.

**Review by manish-agrawal-ai — COMMENTED** [review]
check comment

**Review by manish-agrawal-ai — DISMISSED** [review]
Approved

**Review by manish-agrawal-ai — APPROVED** [review]
approved
